internal/session/repository_session_test.go

```diff
@@ -917,7 +919,47 @@ func TestRepository_CancelSession(t *testing.T) {
 		{
 			name: "already-terminated",
 			session: func() *Session {
-				session := TestDefaultSession(t, conn, wrapper, iamRepo)
+				future := timestamppb.New(time.Now().Add(time.Hour))
+				exp := &timestamp.Timestamp{Timestamp: future}
+				org, proj := iam.TestScopes(t, iamRepo)
+
+				cats := static.TestCatalogs(t, conn, proj.PublicId, 1)
+				hosts := static.TestHosts(t, conn, cats[0].PublicId, 1)
+				sets := static.TestSets(t, conn, cats[0].PublicId, 1)
+				_ = static.TestSetMembers(t, conn, sets[0].PublicId, hosts)
+
+				// We need to set the session connection limit to 1 so that the session
+				// is terminated when the one connection is closed.
```

Comment:
I wonder if the test code isn't sending the going-away RPC that the client does when it's killed. Possibly we'd want to allow either/or but if it was this change wouldn't be necessary.

Reply:
My understanding was that the implicit terminating of a session after a connection is closed is tied to the connection limit of the session, is that wrong? Would we expect the session to terminate when its only connection closes always?